src/app/plot.py

```diff
@@ -0,0 +1,221 @@
+    )
+    
+    bar = alt.Chart(source).mark_bar().encode(
+        x='Age:Q',
+        y='count()',
+    ).properties(
+        width=335,
+        height=170,
+    )
+    
+    fchart = alt.vconcat(chart, bar, spacing=0)
+    
+    return fchart.to_html()
+
+
+def plot_12(xcon):
+    
+    genders = ['Male', 'Female', 'A different identity']
+    if xcon == "World" or xcon is None:
+        source = data[data["GenderSelect"].isin(genders)]
+    else:
+        source = data[(data["Country"] == xcon) & (data["GenderSelect"].isin(genders))]
+        
+    chart = alt.Chart(source).mark_boxplot().encode(
+        x=alt.X("Salary_USD:Q", 
+                title="Salary in USD", 
+                axis=alt.Axis(format='~s'),
+                scale=alt.Scale(zero=False)),
+        y=alt.Y('GenderSelect', title="Gender"),
+        tooltip='count()',
+        color=alt.Color('GenderSelect', title='Gender')
+    ).configure_legend(
+        orient='bottom'
+    ).properties(
+        title='Boxplot by gender',
+        projection={"type":'mercator'},
+        width=575,
+        height=180
+    ).interactive()
+    
+    return chart.to_html()
+
+
+def plot_21(value):
+    
+    education_order = ["Less than bachelor's degree", "Bachelor's degree", 
+                       "Master's degree", "Doctoral degree"]
+    
+    if value == "World" or value is None:
+        country = data
+    else:
+        country = data.query("Country == @value")
+        
+    for idx, i in enumerate(country["FormalEducation"]):
+        if i in education_order[1:]:
+            continue
+        else:
+            print("Change")
+            country["FormalEducation"].iloc[idx] = "Less than bachelor's degree"
+
+    chart = alt.Chart(country).mark_bar().encode(
+            x=alt.X("Salary_USD", bin=alt.Bin(maxbins=20), title="Salary in USD"),
+            y=alt.Y("count()", title="Counts"),
+            color=alt.Color("FormalEducation", sort=education_order,
+            title="Education level"),
+            order=alt.Order('education_order:Q')
+        ).configure_legend(
+            orient='bottom',
+            titleFontSize=8,
+            labelFontSize=8
+        ).properties(
+            title = "Histogram of selected country",
+            width=350,
+            height=180,
+        ).configure_axis(
+            labelFontSize=12
+        )
+
+    return chart.to_html()
+
+
+def plot_11(xmax):
+    
+    world = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
+    world["name"] = world["name"].apply(lambda x:str.lower(" ".join(x.split(" ")[0:2])))
+    
+    world = world.loc[world["name"] != "antarctica"]
+    source = data[["Country", "Salary_USD"]].groupby("Country").median().reset_index()
+    source["Country"] = source["Country"].apply(lambda x:str.lower(x))
+    source.rename({"Country":"name"}, axis=1, inplace=True)
+
+    datamap = pd.merge(world, source, how='left')
+    datamap['Salary_USD'] = datamap['Salary_USD'].fillna(0)
```

Comment:
I think it would be better to just drop `NA`s  than impute them with zero

Reply:
solved: use dropna instead